artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/impl/BridgeImpl.java

```diff
@@ -363,49 +350,35 @@ public Executor getExecutor() {
    }
 
    @Override
-   public void stop() throws Exception {
-      if (stopping) {
+   public synchronized void stop() throws Exception {
+      if (state == State.STOPPING || state == State.STOPPED || state == State.PAUSING) {
```

Comment:
Seems like it would previosuly have tried to stop while already pausing (not saying it would have worked), while this rules out it trying...silently. Feels like throwing e.g ISE to alert of the issue (/ need to try again) might be better if its not actually going to do anything?

Reply:
Agreed. Will fix.

Reply:
Looks like your changes went quite a bit further, also throwing if you start when starting/started, stop when stopping/stopped, or pause when pausing/paused.

Its not really _wrong_, but seems like it could lead to quite a few more exceptions being thrown, and perhaps confusion as to why they were vs previous. Personally I think its ok if those cases nooped, as its already in or heading toward the state being requested. Thats entirely different than what I commented on here, where it was silently ignoring a call to stop due to already pausing (rather than e.g queuing up the stop after the pausing operation) and thus leaving it silently reaching an entirely different state than requested.

Reply:
I made calls to start, stop, & pause idempotent if the state already matches (or will match). If the state doesn't match then an exception is thrown.